image/image_test.go

```diff
@@ -54,6 +55,28 @@ func TestMarshalKeyOrder(t *testing.T) {
 	}
 }
 
+func TestImageWithAllMethods(t *testing.T) {
+	var cid = "50a16564e727"
+	var raw = []byte("279533a6d3a9")
+	config := &container.Config{}
+	var platform = runtime.GOOS
+
+	img := &Image{
+		V1Image: V1Image{
+			Config: config,
+		},
+		rawJSON:    raw,
+		computedID: ID(cid),
+	}
+
+	assert.Equal(t, cid, img.ImageID())
+	assert.Equal(t, cid, img.ID().String())
+	assert.Equal(t, raw, img.RawJSON())
+	assert.Equal(t, platform, img.Platform())
```

Comment:
This tests only the case where `img.OS` is empty, but not if it _is_ set. Perhaps this should in a separate test and it should test both if `img.OS` is set, and `img.OS` is empty (see https://github.com/moby/moby/blob/1daa10301acfe0742dd852dfb8603c48afdaa1ee/image/image.go#L99-L106)